python/lsst/ts/dream/csc/dream_csc.py

```diff
@@ -231,65 +235,73 @@ async def weather_loop(self) -> None:
                     self.log.debug(f"Connected to ESS:{self.config.ess_index}.")
 
                     self.weather_ok_flag = None
-                    self.last_weather_update = 0.0
                     if self.ess_remote is None:
                         self.log.error("Failed to connect to weather CSC.")
                         continue
 
+                    # Wait for the CSC to establish its connection.
+                    await asyncio.sleep(10.0)
```

Comment:
Is it really necessary to wait this long? In any case, please extract the magic number `10.0` to a constant.